stats/record.go

```diff
@@ -58,6 +65,14 @@ func WithMeasurements(measurements ...Measurement) Options {
 	}
 }
 
+// WithMeter records the measurements to the specified `view.Meter`, rather
+// than to the global metrics recorder.
+func WithMeter(meter Recorder) Options {
```

Comment:
Performance of RecordWithOptions WithMeter is 2/3rd of RecordWithOptions without meter
```
BenchmarkRecord8WithMeter-8      	  241356	       270 ns/op	     400 B/op	       4 allocs/op
BenchmarkRecord8WithoutMeter-8   	  335601	       197 ns/op	     368 B/op	       3 allocs/op
```

Reply:
Using the following, I get 231 ns/op, rather than 270 ns/op:

```go
func BenchmarkRecord8_WithRecorder(b *testing.B) {
	ctx := context.Background()
	meter := view.NewMeter()
	meter.Start()
	defer meter.Stop()
	b.ResetTimer()

	for i := 0; i < b.N; i++ {
		stats.RecordWithOptions(ctx, stats.WithRecorder(meter), stats.WithMeasurements(m.M(1), m.M(1), m.M(1), m.M(1), m.M(1), m.M(1), m.M(1), m.M(1)))
	}

	b.StopTimer()
}
```

For the _Parallel case, I get 117 ns/op vs 108 ns/op:

```
goos: windows
goarch: amd64
pkg: go.opencensus.io/stats
BenchmarkRecord0-16                             12499986                98.0 ns/op
BenchmarkRecord1-16                              8571868               139 ns/op
BenchmarkRecord8-16                              6417102               188 ns/op
BenchmarkRecord8_WithRecorder-16                 5194611               231 ns/op
BenchmarkRecord8_WithRecorder_Parallel-16       10085448               117 ns/op
BenchmarkRecord8_Parallel-16                    11215130               108 ns/op
BenchmarkRecord8_8Tags-16                        5940622               198 ns/op
```

I can avoid the extra allocation and make the time between BenchmarkRecord8_WithRecorder and BenchmarkRecord8 match by pre-creating the `stats.WithRecorder(meter)` object like so:

```go
rec := stats.WithRecorder(meter)

for (....) {
  stats.RecordWithOptions(ctx, rec, stats.WithMeasurements(....)
}
```

I note that the current benchmark doesn't perform any aggregation, so it's actually testing the shortcut path AFAICT. Adding a view which is tracking a simple Sum (no tags) increases the benchmark time like so:

```
BenchmarkRecord8_Aggregated-16                   1000000              1017 ns/op
```

Note that using a view here also affects any subsequent benchmarks for the same measurement, because the `m.desc.subscribed()` check around line 120 of record.go will always return true once it has been registered once.

Reply:
BenchmarkRecord8_WithRecorder and BenchmarkRecord8 match by pre-creating the `stats.WithRecorder(meter)` object like so:
> 
> ```go
> rec := stats.WithRecorder(meter)
> 
> for (....) {
>   stats.RecordWithOptions(ctx, rec, stats.WithMeasurements(....)
> }
> ```
> 

+1.